code/modules/transport/pods/secondary_system.dm

```diff
@@ -318,6 +318,18 @@
 	step(C, turn(ship.dir,180))
 	return C
 
+/obj/item/shipcomponent/secondary_system/cargo/proc/get_unloadable(mob/user)
+	var/list/cargo_by_name = list()
+	var/list/counts_by_type = list()
+	for (var/atom/movable/AM as anything in src.load)
+		counts_by_type[AM.type] += 1
+		if (counts_by_type[AM.type] == 1)
+			cargo_by_name[AM.name] = AM
+		else
+			cargo_by_name["[AM.name] #[counts_by_type[AM.type]]"] = AM
+
+	return cargo_by_name[tgui_input_list(user, "Choose which cargo to unload", "Choose Cargo", sortList(cargo_by_name, GLOBAL_PROC_REF(cmp_text_asc)))]
```

Comment:
This will runtime if the user hits cancel, don't assume user input procs will always return nice values.

Reply:
nah i tested that case, no issues!

Reply:
What's returned?

Reply:
> Member

null